client/rest-high-level/src/main/java/org/elasticsearch/client/indices/AnalyzeResponse.java

```diff
@@ -0,0 +1,259 @@
+                    position = parser.intValue();
+                } else if (AnalyzeResponse.Fields.START_OFFSET.equals(field)) {
+                    startOffset = parser.intValue();
+                } else if (AnalyzeResponse.Fields.END_OFFSET.equals(field)) {
+                    endOffset = parser.intValue();
+                } else if (AnalyzeResponse.Fields.POSITION_LENGTH.equals(field)) {
+                    positionLength = parser.intValue();
+                } else if (AnalyzeResponse.Fields.TYPE.equals(field)) {
+                    type = parser.text();
+                } else {
+                    if (t == XContentParser.Token.VALUE_STRING) {
+                        attributes.put(field, parser.text());
+                    } else if (t == XContentParser.Token.VALUE_NUMBER) {
+                        attributes.put(field, parser.numberValue());
+                    } else if (t == XContentParser.Token.VALUE_BOOLEAN) {
+                        attributes.put(field, parser.booleanValue());
+                    } else if (t == XContentParser.Token.START_OBJECT) {
+                        attributes.put(field, parser.map());
+                    } else if (t == XContentParser.Token.START_ARRAY) {
+                        attributes.put(field, parser.list());
+                    }
+                }
+            }
+            return new AnalyzeResponse.AnalyzeToken(term, position, startOffset, endOffset, positionLength, type, attributes);
+        }
+    }
+
+    private final DetailAnalyzeResponse detail;
+    private final List<AnalyzeResponse.AnalyzeToken> tokens;
+
+    public AnalyzeResponse(List<AnalyzeResponse.AnalyzeToken> tokens, DetailAnalyzeResponse detail) {
+        this.tokens = tokens;
+        this.detail = detail;
+    }
+
+    public List<AnalyzeResponse.AnalyzeToken> getTokens() {
+        return this.tokens;
+    }
+
+    public DetailAnalyzeResponse detail() {
+        return this.detail;
+    }
+
+    @Override
+    public Iterator<AnalyzeResponse.AnalyzeToken> iterator() {
+        return tokens.iterator();
+    }
+
+    @Override
+    public XContentBuilder toXContent(XContentBuilder builder, Params params) throws IOException {
+        builder.startObject();
+        if (tokens != null) {
+            builder.startArray(AnalyzeResponse.Fields.TOKENS);
+            for (AnalyzeResponse.AnalyzeToken token : tokens) {
+                token.toXContent(builder, params);
+            }
+            builder.endArray();
+        }
+
+        if (detail != null) {
+            builder.startObject(AnalyzeResponse.Fields.DETAIL);
+            detail.toXContent(builder, params);
+            builder.endObject();
+        }
+        builder.endObject();
+        return builder;
+    }
+
+    @SuppressWarnings("unchecked")
+    private static final ConstructingObjectParser<AnalyzeResponse, Void> PARSER = new ConstructingObjectParser<>("analyze_response",
```

Comment:
This sucks, and there is presumably a better way to deal with lists of objects here?

Reply:
`declareObjectArray` is the way to deal with it, the only critique is that the item transformation (the `AnalyzeToken`) is not using another parser, like how the `DetailAnalyzeResponse.PARSER` is used in the next line.